src/ServiceProvider/Kernel/ServiceProviderSpecimenBuilder.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using AutoFixture.Kernel;
+
+namespace Kralizek.AutoFixture.Extensions.Kernel;
+
+public class ServiceProviderSpecimenBuilder : ISpecimenBuilder
+{
+    private readonly IServiceProvider _serviceProvider;
+
+    public ServiceProviderSpecimenBuilder(IServiceProvider serviceProvider)
+    {
+        _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+    }
+    
+    public object Create(object request, ISpecimenContext context)
+    {
+        _ = request ?? throw new ArgumentNullException(nameof(request));
```

Comment:
These guard expressions are pretty unusual. In my view they draw too much attention when you read the code.
However if this your preference and it makes the code more readable for you then keep it.

Reply:
They definitely take less space than a full blown `if(){}` block. The new `ArgumentNullException.ThrowIfNull` solves the issue but is only available in .NET 6 and above.